### PR TITLE
EIP-7742 Engine API Changes

### PR DESCRIPTION
**Pull request by siladu**
## PR description

**This makes Pectra require EIP-7742 fields so will break any pectra devnets that exclude 7742.** @daniellehrner will that be an issue? I could temporarily disable the engine api validation to avoid this if so.

This makes Besu usable for interop (has been tested with lodestar so far), however there are still some missing pieces that may cause 7742 to break depending on how maxBlobsPerBlock is set:
* The max isn't wired through to block creation so that still assumes Cancun's 6 blobs max.
* The 4844 block/tx max blob validation (via newPayload) hasn't been skipped for Prague so a newPayload with > 6 blobs may fail

In other words, in order for Besu to work for a Pectra interop we would require a EIP-7742 compatible CL and a max blob value of <= 6.

This PR was big enough already and I think uncontentious, whereas the implementations for the points above may require more discussion.

Also block creation wiring was required in this PR in order to satisfy CodeDelegationTransactionAcceptanceTest

Part of https://github.com/hyperledger/besu/issues/7605

Spec: https://github.com/ethereum/execution-apis/pull/574

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/PayloadIdentifier.java [hunk 1]
```diff
@@ -71,23 +72,26 @@ public static PayloadIdentifier forPayloadParams(
...
-            ^ ((long) parentBeaconBlockRoot.hashCode()) << 40);
+                    withdrawals
+                        .map(
+                            ws ->
+                                ws.stream()
+                                    .sorted(Comparator.comparing(Withdrawal::getIndex))
+                                    .map(Withdrawal::hashCode)
+                                    .reduce(1, (a, b) -> a ^ (b * 31)))
+                        .orElse(0)
+                << 32
```
Comment (siladu) [inline consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/PayloadIdentifier.java:92]: Note, added a shift for withdrawals to keep the hashcodes evenly distributed by using the same pattern as the other fields.

**Comment by daniellehrner** [comment]
@siladu Breaking existing testnets isn't an issue. Mekong will not be update, it's spec stays frozen, and devnet-4 has been shut down already

File: consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/PayloadIdentifier.java (hunk 1 above)
Comment (macfarla) [reply consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/PayloadIdentifier.java:94]: is it worth extracting this to a method? it's a bit hard to read indented over so many lines

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java [hunk 2]
```diff
@@ -318,6 +320,10 @@ private void logPayload(final EnginePayloadAttributesParameter payloadAttributes
               .setMessage(message)
               .addArgument(() -> payloadAttributes.getParentBeaconBlockRoot().toHexString());
     }
+    if (payloadAttributes.getTargetBlobsPerBlock() != null) {
+      message += ", targetBlobsPerBlock: {}";
+      builder = builder.setMessage(message).addArgument(payloadAttributes::getTargetBlobsPerBlock);
+    }
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineForkchoiceUpdated.java:326]: why not log getMaxBlobsPerBlock here too

**Comment by siladu** [comment]
Removed from Pectra

**Pull request closed without merging**